owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/shares/UpdateRemoteShareOperation.kt

```diff
@@ -0,0 +1,232 @@
+ *   furnished to do so, subject to the following conditions:
+ *
+ *   The above copyright notice and this permission notice shall be included in
+ *   all copies or substantial portions of the Software.
+ *
+ *   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ *   EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ *   MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ *   NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ *   BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ *   ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ *   CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ *   THE SOFTWARE.
+ *
+ */
+
+package com.owncloud.android.lib.resources.shares
+
+import com.owncloud.android.lib.common.OwnCloudClient
+import com.owncloud.android.lib.common.http.HttpConstants
+import com.owncloud.android.lib.common.http.methods.nonwebdav.PutMethod
+import com.owncloud.android.lib.common.operations.RemoteOperation
+import com.owncloud.android.lib.common.operations.RemoteOperationResult
+import com.owncloud.android.lib.common.utils.Log_OC
+import okhttp3.FormBody
+import java.net.URL
+import java.text.SimpleDateFormat
+import java.util.Calendar
+import java.util.Locale
+
+/**
+ * Updates parameters of an existing Share resource, known its remote ID.
+ *
+ *
+ * Allow updating several parameters, triggering a request to the server per parameter.
+ *
+ * @author David A. Velasco
+ * @author David González Verdugo
+ */
+
+class UpdateRemoteShareOperation
+/**
+ * Constructor. No update is initialized by default, need to be applied with setters below.
+ *
+ * @param remoteId Identifier of the share to update.
+ */
+    (
+    /**
+     * Identifier of the share to update
+     */
+    private val mRemoteId: Long
+) : RemoteOperation<ShareParserResult>() {
+
+    /**
+     * Password to set for the public link
+     */
+    private var mPassword: String? = null
+
+    /**
+     * Expiration date to set for the public link
+     */
+    private var mExpirationDateInMillis: Long = 0
+
+    /**
+     * Access permissions for the file bound to the share
+     */
+    private var mPermissions: Int = 0
+
+    /**
+     * Upload permissions for the public link (only folders)
+     */
+    private var mPublicUpload: Boolean? = null
+    private var mName: String? = null
+
+    init {
+        mPassword = null               // no update
+        mExpirationDateInMillis = 0    // no update
+        mPublicUpload = null
+        mPermissions = RemoteShare.DEFAULT_PERMISSION
+    }
+
+    /**
+     * Set name to update in Share resource. Ignored by servers previous to version 10.0.0
+     *
+     * @param name Name to set to the target share.
+     * Empty string clears the current name.
+     * Null results in no update applied to the name.
+     */
+    fun setName(name: String) {
+        this.mName = name
```

Comment:
I don't know why you are using m prefix and custom setters again...

Reply:
It does not make any sense but is exactly what Android Studio Java to Kotlin converter did. I'd rather trust on this converter because reviewing whole classes will be crazy, let's see. Anyway, this is already fixed.